src/vs/workbench/api/node/extHost.protocol.ts

```diff
@@ -193,6 +193,7 @@ export interface MainThreadTextEditorsShape extends IDisposable {
 	$tryShowEditor(id: string, position: EditorViewColumn): TPromise<void>;
 	$tryHideEditor(id: string): TPromise<void>;
 	$trySetOptions(id: string, options: ITextEditorConfigurationUpdate): TPromise<void>;
+	$trySetLanguageById(id: string, languageId: string): TPromise<void>;
```

Comment:
Consider moving this `ExtHostLanguages`, esp given below considerations. Then, the change shouldn't work for an editor id but just for a document uri.